### PR TITLE
fix tracking

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?

1. When dictionary of metrics is provided to the accelerate tracker, tensorboard wasn't able to log it. Example with dict of metrics is in this transformers script [run_glue_no_trainer.py](https://github.com/huggingface/transformers/blob/9f16a1cc13029c37fd3b7713d0edb743a8a86e39/examples/pytorch/text-classification/run_glue_no_trainer.py#L541). Fixed support for dictionary of int/float metrics 
2. Fixing examples so that run is created only for the main process else wandb will create num_processes runs with no data.
3. comet-ml logging wasn't helpful as everything was being logged in others `log_others`. No plots were being created. Fixing it to be able to get plots for metrics using `log_metric` and `log_metrics`.

Post these changes, below are the plots for the [run_glue_no_trainer.py](https://github.com/huggingface/transformers/blob/main/examples/pytorch/text-classification/run_glue_no_trainer.py) script on MRPC task using all trackers.
Screenshot 2022-05-12 at 8 11 16 PM
Screenshot 2022-05-12 at 8 10 51 PM
Screenshot 2022-05-12 at 8 10 17 PM

### CODE REVIEW

File: examples/by_feature/fsdp_with_peak_mem_tracking.py [hunk 1]
```diff
@@ -293,7 +294,7 @@ def collate_fn(examples):
                     {
                         "accuracy": eval_metric["accuracy"],
                         "f1": eval_metric["f1"],
-                        "train_loss": total_loss,
+                        "train_loss": total_loss.item() if type(total_loss) == torch.Tensor else total_loss,
```
Comment (sgugger) [inline examples/by_feature/fsdp_with_peak_mem_tracking.py:297]: Do we really have either a Tensor or something else? Can't it always be `.item()` here?

File: examples/by_feature/tracking.py [hunk 2]
```diff
@@ -208,15 +209,16 @@ def training_function(config, args):
...
         # To actually log, we call `Accelerator.log`
-        # The values passed can be of `str`, `int`, or `float`
+        # The values passed can be of `str`, `int`, `float` or `dict` of `str` to `float`/`int`
         if args.with_tracking:
             accelerator.log(
                 {
                     "accuracy": eval_metric["accuracy"],
                     "f1": eval_metric["f1"],
-                    "train_loss": total_loss,
+                    "train_loss": total_loss.item() if type(total_loss) == torch.Tensor else total_loss,
```
Comment (sgugger) [inline examples/by_feature/tracking.py:218]: Same comment

**Review by sgugger — COMMENTED** [review]
Thanks for looking into this!

**Comment by sgugger** [comment]
Oh, also make sure to edit the title of the PR ;-)

**Review by muellerzr — APPROVED** [review]
LG2M, will handle looking at the failing diff test tomorrow morning, as not sure quite what's going on there.

**Pull request merged**